### PR TITLE
base64: add base64url support

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <zlizan@google.com>

*Description*:
Add [base64url](https://tools.ietf.org/html/rfc4648#section-5) support in common utils, this is a dependency of JWT #2514 support.
*Risk Level*: Low
*Testing*: unit test
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

**Comment by zuercher** [comment]
Codewise, this seems fine. Is there a standard for this base64 variant that we can reference in the docs?

**Comment by lizan** [comment]
[RFC4648](https://tools.ietf.org/html/rfc4648) is the one, I'll add it to comment in header.

File: source/common/common/base64.cc [hunk 1]
```diff
@@ -24,118 +27,149 @@ static const unsigned char REVERSE_LOOKUP_TABLE[256] = {
     64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64,
     64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64};
 
-std::string Base64::decode(const std::string& input) {
-  if (input.length() % 4 || input.empty()) {
-    return EMPTY_STRING;
+constexpr char URL_CHAR_TABLE[] =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";
+
+constexpr unsigned char URL_REVERSE_LOOKUP_TABLE[256] = {
```
Comment (mattklein123) [inline source/common/common/base64.cc]: How was this table generated? Did you copy from the same place as above? If so an you comment here also?

**Comment by zuercher** [comment]
Add base64url support in common utils, this is a dependency of JWT #2514 support.

Risk Level: Low
Testing: unit test
Release Notes: N/A

Signed-off-by: <Lizan Zhou zlizan@google.com>

**Pull request merged**